File: BeitPatissierServer/Models/ProductTag.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace BeitPatissierServer.Models
{
    public class ProductTag //כגון בר מצווה, בוקר עובדים, יום הולדת
    {
        public int ProductId { get; set; }
        [ForeignKey(nameof(ProductId))]
        public Product Product { get; set; }

        public int TagId { get; set; }
        [ForeignKey(nameof(TagId))]
        public Tag Tag { get; set; }
    }
}

File: beit-patissier-manage/src/models/ProductTag.ts
import { Product } from './Product';
import { Tag } from './Tag';

export class ProductTag {
    productId: number;
    product: Product;
    tagId: number;
    tag: Tag;

    constructor(
        productId: number,
        product: Product,
        tagId: number,
        tag: Tag
    ) {
        this.productId = productId;
        this.product = product;
        this.tagId = tagId;
        this.tag = tag;
    }
}
